stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -692,15 +692,14 @@ extension Unicode.Scalar.Properties {
   /// all current case mappings. In the event more space is needed, it will be
   /// allocated on the heap.
   internal func _applyMapping(_ u_strTo: _U_StrToX) -> String {
-    // TODO(String performance): Stack buffer first and then detect real count
     let count = 64
-    var array = Array<UInt16>(repeating: 0, count: count)
+    var array = _FixedArray16<UInt64>(allZeros: ()) // 16 * UInt64 take 128 bytes, same as 64 * UInt16
     let len: Int = array.withUnsafeMutableBufferPointer { bufPtr in
       return _scalar.withUTF16CodeUnits { utf16 in
         var err = __swift_stdlib_U_ZERO_ERROR
         let correctSize = u_strTo(
-          bufPtr.baseAddress._unsafelyUnwrappedUnchecked,
-          Int32(bufPtr.count),
+          UnsafeMutableRawPointer(bufPtr.baseAddress!).bindMemory(to: UInt16.self, capacity: count),
```

Comment:
This binds the memory to `UInt16` but doesn't bind it back to `UInt64`, as later accesses would expect.

Reply:
Aren't these bindings noops in case of primitive types?

Reply:
I don't believe we guarantee that to be the case.

Reply:
fixed